lms/services/file.py

```diff
@@ -55,6 +55,23 @@ def upsert(self, file_dicts):
             update_columns=["name", "size", "updated"],
         )
 
+    def _file_search_query(
+        self, application_instance, type_, *, lms_id=None, course_id=None
+    ):
+        """Return a `File` query with the passed parameters applied as filters."""
+        query = self._db.query(File).filter_by(
```

Comment:
We always query by these two parameters. I reckon always filtering by them is better than making them optional and having to test the private method.

If we have need to make them optional is a quick and obvious change.